.reaction/docker/scripts/build-meteor.sh

```diff
@@ -1,30 +1,35 @@
 #!/bin/bash
 
 #
-# builds a production meteor bundle directory
+# builds a production meteor bundle
 #
 set -e
 
-printf "\n[-] Building Meteor application...\n\n"
-
-: ${APP_BUNDLE_DIR:="/var/www"}
-: ${BUILD_SCRIPTS_DIR:="/opt/reaction"}
-
 cd $APP_SOURCE_DIR
 
 # Customize packages
-bash $BUILD_SCRIPTS_DIR/build-packages.sh
+$BUILD_SCRIPTS_DIR/build-packages.sh
 
 # Generate plugin import files
-bash $BUILD_SCRIPTS_DIR/plugin-loader.sh
+printf "\n[-] Running Reaction plugin loader...\n\n"
+$BUILD_SCRIPTS_DIR/plugin-loader.sh
 
 # Install app deps
+printf "\n[-] Running npm install...\n\n"
 meteor npm install
 
-# build the source
+# build the production bundle
+printf "\n[-] Building Meteor application...\n\n"
 mkdir -p $APP_BUNDLE_DIR
-meteor build --directory $APP_BUNDLE_DIR
-cd $APP_BUNDLE_DIR/bundle/programs/server/ && meteor npm install --production
+meteor build --unsafe-perm --directory $APP_BUNDLE_DIR
```

Comment:
The app is built as root because there were a lot of random issues in the build process (depending on 3rd party packages a user may have).  As soon as the build is done, the permissions of the entire directory are changed to non-root and the app is started as that same user.  

Building as non-root would sometimes fail because of permissions issues caused by some npm packages and building as root would fail because Meteor 1.4.2 won't allow it without that --unsafe-perm flag.  I assure you, this was the least awful solution until Meteor releases 1.4.2.1 with a better answer for --unsafe-perm (already fixed, just waiting for release).  

In the end, this is accomplishing the exact same thing while avoiding fussy permissions on some module builds.